conan/tools/system/package_manager.py

```diff
@@ -56,9 +56,11 @@ def get_default_tool(self):
                 if d in os_name:
                     return tool
 
-    def get_package_name(self, package):
-        # TODO: should we only add the arch if cross-building?
-        if self._arch in self._arch_names and cross_building(self._conanfile):
+    def get_package_name(self, package, host_tool=True):        
+        # Only if the package is for building, for example a library,
+        # we should add the destination arch when cross building.
+        # If the package is a tool that should be installed on the host we should not add the arch.
+        if self._arch in self._arch_names and cross_building(self._conanfile) and not host_tool:
```

Comment:
I think that the naming might be the opposite. We are using GNU, which means:
- "build" is the current build machine
- "host" is the machine that will run the software
If we are cross-building in a X86_64 box for an armv8 target, our "build" is x86_64 and the "host" is "armv8".
Then, I understand that the "mypkg:arm" qualifier needs to be added when it is a "host_package" (I'd probably avoid the ``_tool`` name here, as there are both libraries and tools in system packages), so maybe the semantics are reversed?

Reply:
Ops, I refer to the host machine as the current build machine, I will change it.

Reply:
I have changed also in the docs repo